tests/net/socket/offload_dispatcher/src/main.c

```diff
@@ -842,65 +844,39 @@ static void test_tls_native_iface_native(void)
-					       test_socket_teardown),
-		ztest_unit_test_setup_teardown(test_sendmsg_not_bound,
-					       test_socket_setup_udp,
-					       test_socket_teardown),
-		ztest_unit_test_setup_teardown(test_getpeername_not_bound,
-					       test_socket_setup_udp,
-					       test_socket_teardown),
-		ztest_unit_test_setup_teardown(test_getsockname_not_bound,
-					       test_socket_setup_udp,
-					       test_socket_teardown),
-		ztest_unit_test_setup_teardown(test_so_bindtodevice_iface_offloaded,
-					       test_socket_setup_udp,
-					       test_socket_teardown),
-		ztest_unit_test_setup_teardown(test_so_bindtodevice_iface_native,
-					       test_socket_setup_udp,
-					       test_socket_teardown),
-		ztest_unit_test_setup_teardown(test_tls_native_iface_offloaded,
-					       test_socket_setup_tls,
-					       test_socket_teardown),
-		ztest_unit_test_setup_teardown(test_tls_native_iface_native,
-					       test_socket_setup_tls,
-					       test_socket_teardown)
-	);
-
-	ztest_run_test_suite(socket_offload_dispatcher);
-}
+ZTEST_SUITE(socket_offload_dispatcher_close_test_suite, NULL, test_socket_setup_udp,
+	    NULL, NULL, NULL);
+ZTEST_SUITE(socket_offload_dispatcher_fcntl_test_suite, NULL, test_socket_setup_udp,
+	    NULL, NULL, test_socket_teardown);
+ZTEST_SUITE(socket_offload_dispatcher_shutdown_test_suite, NULL, test_socket_setup_udp,
+	    NULL, NULL, test_socket_teardown);
+ZTEST_SUITE(socket_offload_bind_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_connect_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_listen_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_accept_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_sendto_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_recvfrom_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getsockopt_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_setsockopt_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_sendmsg_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getpeername_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getsockname_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_so_offload_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_so_native_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
```

Comment:
These all have the exact same setup/teardown, why can't we just use 1 suite?

Reply:
only can combine to 3 suite